registers/dump/main.go

```diff
@@ -1,22 +1,69 @@
 package main
 
 import (
+	"context"
+	"encoding/json"
 	"flag"
+	"fmt"
 	"log"
 
 	"github.com/benlaurie/gds-registers/register"
+	"github.com/google/trillian"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
 )
 
 var (
-	regName = flag.String("register", "register", "name of register (e.g. 'country')")
+	regName     = flag.String("register", "register", "name of register (e.g. 'country')")
+	trillianLog = flag.String("trillian_log", "localhost:8090", "address of the Trillian Log RPC server.")
+	logID       = flag.Int64("log_id", 0, "Trillian LogID to populate.")
 )
 
 type dumper struct {
+	tc               trillian.TrillianLogClient
+	logID            int64
+	ctx              context.Context
+	newEntries       uint64
+	duplicateEntries uint64
+}
+
+type leaf struct {
+	Entry map[string]interface{}
+	Hash  string
+	Item  map[string]interface{}
 }
 
 // Process implements the register.ItemProcessor interface.
-func (*dumper) Process(e map[string]interface{}, h string, i map[string]interface{}) error {
+func (d *dumper) Process(e map[string]interface{}, h string, i map[string]interface{}) error {
 	log.Printf("%#v %s %#v", e, h, i)
+
+	// Put all three parts in a single structure and serialise to JSON
+	l := leaf{Entry: e, Hash: h, Item: i}
+	j, err := json.Marshal(l)
+	if err != nil {
+		return err
+	}
+
+	// Send to Trillian
+	tl := &trillian.LogLeaf{LeafValue: j}
+	q := &trillian.QueueLeafRequest{LogId: d.logID, Leaf: tl}
+	r, err := d.tc.QueueLeaf(d.ctx, q)
```

Comment:
Some of those short variable names can be hard to understand (ex. is this `r` for request or response?).

I prefer renaming some of them to make them more readable.

Reply:
The golang style prefers short identifiers, see [here](https://github.com/golang/go/wiki/CodeReviewComments#variable-names)